cli/src/merge_tools/builtin.rs

```diff
@@ -380,7 +423,33 @@ pub fn apply_diff_builtin(
         let (selected, _unselected) = file.get_selected_contents();
         match selected {
             scm_record::SelectedContents::Absent => {
-                tree_builder.set_or_remove(path, Merge::absent());
+                // TODO(https://github.com/arxanas/scm-record/issues/26): This
+                // is probably an upstream bug in scm-record. If the file exists
+                // but is empty, `get_selected_contents` should probably return
+                // `Unchanged` or `Present`. When this is fixed upstream we can
+                // simplify this logic.
+
+                // Currently, `Absent` means the file is either empty or
+                // deleted. We need to disambiguate three cases:
+                // 1. The file is new and empty.
+                // 2. The file existed before, is empty, and nothing changed.
```

Comment:
comment (non-blocking): I imagine that we shouldn't/don't generate any files for the UI in case 2, when the file hasn't changed?

Reply:
I think what I mean here is that the user didn't select any changes, therefore the file won't have changed in the resulting tree. I _think_ that after the upstream changes the three cases will be mapped like this:

1. The file is new and empty => `Present("")`
2. The file existed before, is empty, and nothing changed => `Unchanged`
3. The file does not exist (it's been deleted) => `Absent`

I'm working on a patch for the upstream UI and will probably change `get_selected_contents` so that it behaves this way and disambiguates these cases. Assuming that gets merged we can update the dep for `jj` and clean up this `Absent` block :)